proto/buf/alpha/registry/v1alpha1/labels.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+enum LabelNamespace {
+  LABEL_NAMESPACE_UNSPECIFIED = 0;
+  // This is for globs, '*', in requests. When a label is created
+  // it will be assigned to a specific namespace.
+  LABEL_NAMESPACE_ALL = 1;
+  LABEL_NAMESPACE_TAG = 2;
+  LABEL_NAMESPACE_BRANCH = 3;
+  LABEL_NAMESPACE_GITCOMMIT = 4;
+}
+
+message Label {
+  LabelNamespace namespace = 1;
+  string name = 2;
+  string commit_id = 3;
+}
+
+service LabelService {
+  rpc CreateLabel(CreateLabelRequest) returns (CreateLabelResponse);
+  rpc MoveLabel(MoveLabelRequest) returns (MoveLabelResponse);
+  rpc GetLabelByNameAndNamespace(GetLabelByNameAndNamespaceRequest) returns (GetLabelByNameAndNamespaceResponse);
```

Comment:
This is a misnomer if namespace is committed, which we should support. E.g. this should be a valid call:

```json
{"name": "main"}
```

`GetLabel` should be sufficient since the request specifies what the parameters are.

```proto
rpc GetLabel(GetLabelRequest) returns (GetLabelResponse);
```

Reply:
Also, is there no circumstance where we'd want to query for a name within a specific namespace? If we are always only ever retrieving by name, we are priced into querying all labels in all namespaces always?

Reply:
> we would not be providing a namespace for querying, which also means we won't need the ALL namespace, right?

Yep, we can use omission to mean all.

> there no circumstance where we'd want to query for a name within a specific namespace? If we are always only ever retrieving by name, we are priced into querying all labels in all namespaces always?

I believe both kinds of calls will be made. A plain name would be often queried, but that shouldn't exclude including a namespace. Similarly to getting a single name, this can support getting a /namespace/. E.g. to list all tags:

```json
{"namespace": "LABEL_NAMESPACE_TAG"}
```

Reply:
Yeah, this makes sense -- but also makes sense that we're doing `GetLabels` :)